docs/Data-flow-in-Site-Kit.md

```diff
@@ -1,3 +1,155 @@
+
+Generally speaking, we cache report data that does not vary (either much or at all) every hour.
+
+If the WordPress REST API returns a non-`2xx` HTTP response code (or the request fails entirely), the client will not save any data for this request in a cache and will instead log the error using a Redux (`@wordpress/data`) action. Usually the client will display this error in an appropriate fashion. Again: failed request data is never cached in `sessionStorage`.
+
+If the data was retrieved from the cache, the cache will not be modified. Cached data is stored with a time-to-live and will be cleared/ignored if "stale".
+
+## Creating a fetch action/selector using `createFetchStore`
+
+When creating an action/selector that triggers a `fetch` request, use the `createFetchStore` function to return a new [data store that can be combined with an existing store](./Data-store-architecture.md).
+
+Here's an example of what a `createFetchStore` looks like, using the existing `core/site/data/connection` route as a reference:
+
+```js
+const fetchGetConnectionStore = createFetchStore( {
+ baseName: 'getConnection',
+ // Make the API request; by default the cache is used for `get` calls.
+ controlCallback: () => {
+  return API.get( 'core', 'site', 'connection' );
+ },
+ // Changes to make to the reducer once the control finishes and gets a response.
+ reducerCallback: ( state, connection ) => {
+  return {
+   ...state,
+   connection,
+  };
+ },
+} );
+```
+
+## Defining REST API endpoints on the server
+
+REST API endpoints are defined for each module, using four methods for the `Module` class. These cover:
+
+1. Set up remote requests (`setup_services`)
+2. REST API routes (`get_datapoint_definitions`)
+3. Data request to make to a 3rd-party API, if needed (`create_data_request`)
+4. Response parsing for the optional data request (`parse_data_response`)
+
+### Set up remote requests
+
+Each module can set up the services that will be used to make third-party requests. The `setup_services` method returns an array of service identifier to service object instances. These "service objects" can be used to interact with a third-party API.
+
+The identifier is referenced in the datapoint definitions in `get_datapoint_definitions` and the service instance when making the request in `create_data_request`. Here is an example `setup_services` implementation from the Search Console module.
+
+```php
+  protected function setup_services( Google_Site_Kit_Client $client ) {
+    return array(
+      'searchconsole' => new Google_Service_SearchConsole( $client ),
+    );
+  }
+```
+
+### REST API routes
+
+Define a module's REST routes using the `get_datapoint_definitions` function. This function defines the array of REST routes using an array where the keys are the REST routes and the values are requirements/properties of that route. Here's an example from Search Console's `get_datapoint_definitions`:
```

Comment:
This is mostly accurate, although datapoints are just one part of a module's rest route, the rest of it identifies the type `modules`, identifier (module slug), and then the datapoint (these).

Others such as `settings` are handled by core infrastructure and use a similar route and aren't considered "datapoints" like these as they also don't proxy to external APIs.

Reply:
I think from the code and explanations the point that a module's `slug` will be part of the REST route is clear.

I think adding that context will probably crowd this section. There's lots to talk about in terms of the intricacies of our REST routes but I wanted this doc will get people started without too many detours 🙂